trainaspower/trainasone.py

```diff
@@ -154,19 +154,29 @@ def convert_steps(steps, config: models.Config, perceived_effort: bool) -> Gener
                 # Provide a generous power range based on %CP for slower ranges
                 if step["targetType"] == "OPEN":
                     if perceived_effort:
-                        if step["stepOrder"] == 2:
-                            # Perceived effort warmup
-                            cp = get_critical_power()
-                            out_step.power_range = models.PowerRange(
-                                cp * 0.3, cp * 0.8)
-                        elif step["stepOrder"] == 3:
-                            # Perceived effort main body
-                            cp = get_critical_power()
-                            out_step.power_range = models.PowerRange(
-                                cp * 0.55, cp * 0.9)
+                        if len(steps) == 3:
+                            if step["stepOrder"] == 2:
+                                # Perceived effort main body
+                                cp = get_critical_power()
+                                out_step.power_range = models.PowerRange(
+                                    cp * 0.55, cp * 0.9)
+                            else:
+                                # Standing
+                                out_step.power_range = models.PowerRange(0, 50)
                         else:
-                            # Standing
-                            out_step.power_range = models.PowerRange(0, 50)
+                            if step["stepOrder"] == 2:
+                                # Perceived effort warmup
+                                cp = get_critical_power()
+                                out_step.power_range = models.PowerRange(
+                                    cp * 0.3, cp * 0.8)
+                            elif step["stepOrder"] == 3:
+                                # Perceived effort main body
+                                cp = get_critical_power()
+                                out_step.power_range = models.PowerRange(
+                                    cp * 0.55, cp * 0.9)
+                            else:
+                                # Standing
+                                out_step.power_range = models.PowerRange(0, 50)
```

Comment:
Can we avoid repeating the same code for both of these blocks? Maybe something like:
```python
if len(steps) > 3 and step["stepOrder"] == 2:
    # Perceived effort warmup
elif step["stepOrder"] == len(steps) - 1:
    # Perceived effort main body
else:
    # Standing
```

Reply:
Yes you are right; PR updated as suggested